File: Tease AI/Services/TauntAccessor.vb
Option Strict On
Option Infer Off
Imports System.IO
Imports TeaseAI.Common
Imports TeaseAI.Common.Interfaces.Accessors

Public Class TauntAccessor
    Implements ITauntAccessor

    Public Function GetVocabulary(session As Session, keyword As String) As Result(Of List(Of String)) Implements ITauntAccessor.GetTaunt

        Dim filePath As String = Application.StartupPath & "\Scripts\" & session.Domme.PersonalityName & "\Stroke\Edge\Edge.txt"
        'If session.GlitterTease Then
        'filePath = Application.StartupPath & "\Scripts\" & session.Domme.PersonalityName & "\Stroke\Edge\GroupEdge.txt"
        'End If
        If (Not File.Exists(filePath)) Then
            Return Result.Fail(Of List(Of String))("Unable to find vocabulary for " + keyword)
        End If

        Return Result.Ok(File.ReadAllLines(filePath).ToList())
    End Function
End Class

File: TeaseAI.Services/Accessors/TauntAccessor.cs
using System;
using System.Collections.Generic;
using TeaseAI.Common;
using TeaseAI.Common.Interfaces.Accessors;

namespace TeaseAI.Services.Accessors
{
    public class TauntAccessor : ITauntAccessor
    {
        private IConfigurationAccessor _configurationAccessor;

        public TauntAccessor(IConfigurationAccessor configurationAccessor)
        {
            _configurationAccessor = configurationAccessor;
        }

        public Result<List<string>> GetTaunt(Session session, string keyword)
        {
            throw new NotImplementedException();
        }
    }
}
